### ISSUE TITLE
4-bit QLoRA training diverges when using FlashAttention 2

### ISSUE THREAD

**Issue by rationalism** [TimDettmers/bitsandbytes#663]
Model: Llama-2-70B
CUDA: 11.8
bitsandbytes version: 0.41.0
PyTorch version: 2.0.1
OS version: Ubuntu 22.04
GPU: nVidia A6000 (the Ampere one)
Context window: 4,096
Learning rate: 1e-4
Optimizer: 8-bit Adam

Fine-tuning Llama 2 with Flash Attention 2 diverges when done with 4-bit QLoRA, presumably due to some kind of numerical instability. Training appears to work initially, but after ~20 steps, loss starts diverging and getting larger instead of smaller. Setting the learning rate to 0 fixes it (but of course doesn't allow for fine-tuning). Here's the code I'm using:

https://huggingface.co/togethercomputer/LLaMA-2-7B-32K/blob/main/modeling_flash_llama.py

(note that I got a bug due to the JIT decorator on the rmsnorm function, which apparently didn't play nicely with PyTorch's type conversion; removing the decorator fixed this)

**Comment by rationalism** [comment]
This has been fixed by 0.41.1, closing out.
